crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -153,6 +153,17 @@ impl<'w> EntityRef<'w> {
         // SAFETY: We have read-only access to all components of this entity.
         unsafe { self.0.get_by_id(component_id) }
     }
+
+    /// Returns read-only components for the current entity that match the query `Q`.
+    pub fn components<Q: ReadOnlyQueryData>(&self) -> Q::Item<'w> {
+        self.get_components::<Q>().expect(QUERY_MISMATCH_ERROR)
+    }
+
+    /// Returns read-only components for the current entity that match the query `Q`.
+    pub fn get_components<Q: ReadOnlyQueryData>(&self) -> Option<Q::Item<'w>> {
```

Comment:
Should add some docs to explain why it might return `None`. Ditto for the other methods that return an option.